editor.planx.uk/src/@planx/components/FileUploadAndLabel/SelectMultiple.tsx

```diff
@@ -1,233 +1,318 @@
-
-  const handleChange = (event: SelectChangeEvent<typeof tags>) => {
-    const {
-      target: { value },
-    } = event;
-    setTags(
-      // On autofill we get a stringified value.
-      typeof value === "string" ? value.split(",") : value,
-    );
-  };
-  const handleClose = () => setOpen(false);
-  const handleOpen = () => setOpen(true);
 
-  const updateFileListWithTags = (
-    previousTags: string[] | undefined,
-    tags: string[],
+  /**
+   * Options for autocomplete
+   * FileList with appended "category" property for grouping
+   */
+  const options: Option[] = useMemo(
+    () =>
+      (Object.keys(fileList) as Array<keyof typeof fileList>)
+        .filter((fileListCategory) => fileList[fileListCategory].length > 0)
+        .flatMap((category) =>
+          fileList[category].map((fileType) => ({ category, ...fileType })),
+        ),
+    [fileList],
+  );
+
+  /**
+   * Previous values to pre-populate autocomplete
+   */
+  const value: Option[] = useMemo(
+    () =>
+      initialTags.flatMap((tag) => options.filter(({ name }) => name === tag)),
+    [initialTags],
+  );
+
+  /**
+   * Update FileList (from main Public.tsx component) when values are added or removed from Autocomplete
+   */
+  const handleChange = (
+    _event: React.SyntheticEvent,
+    value: Option[],
+    reason: AutocompleteChangeReason,
   ) => {
-    const updatedTags = tags.filter((tag) => !previousTags?.includes(tag));
-    const removedTags = previousTags?.filter((tag) => !tags?.includes(tag));
-
-    if (updatedTags.length > 0) {
-      const updatedFileList = addOrAppendSlots(
-        updatedTags,
-        uploadedFile,
-        fileList,
-      );
-      setFileList(updatedFileList);
-    }
+    const selectedTags = value.map(({ name }) => name);
 
-    if (removedTags && removedTags.length > 0) {
-      const updatedFileList = removeSlots(removedTags, uploadedFile, fileList);
-      setFileList(updatedFileList);
+    switch (reason) {
+      case "selectOption": {
+        const updatedTags = selectedTags.filter(
+          (tag) => !initialTags?.includes(tag),
+        );
+        const updatedFileList = addOrAppendSlots(
+          updatedTags,
+          uploadedFile,
+          fileList,
+        );
+        setFileList(updatedFileList);
+        break;
+      }
+      case "removeOption": {
+        const removedTags = initialTags?.filter(
+          (tag) => !selectedTags?.includes(tag),
+        );
+        const updatedFileList = removeSlots(
+          removedTags,
+          uploadedFile,
+          fileList,
+        );
+        setFileList(updatedFileList);
+        break;
+      }
     }
   };
 
-  useEffect(() => {
-    updateFileListWithTags(previousTags, tags);
-  }, [tags]);
+  /**
+   * Custom Listbox component
+   * Used to wrap options within the autocomplete and append a custom element above the option list
+   */
+  const ListboxComponent = forwardRef<typeof Box, PropsWithChildren>(
```

Comment:
I spent a fair bit of time trying to take this component out of the scope of `SelectMultiple` and pass in the `closePopper()` function as a prop without much luck.